opentech/apply/projects/views.py

```diff
@@ -104,6 +129,38 @@ def percentage(total, value):
         }
 
 
+class SubmissionFilesMixin:
+    """
+    Mixin to provide an instantiated SelectDocumentForm
+    """
+    def get_context_data(self, **kwargs):
+        project = self.get_object()
+
+        paths = self.get_file_paths(project)
+
+        context = super().get_context_data(**kwargs)
+        context['select_document_form'] = SelectDocumentForm(paths, project)
+        return context
+
+    def get_file_paths(self, project):
+        """
+        Get MEDIA_ROOT-relative paths for all files linked to a Project's Submission
+        """
+        # Get field instances as an iterable
+        fields = project.submission.form_data.values()
+
+        # Make sure that list is flat.  We specify the base_type here so
+        # collapse doesn't try to iterate over the StreamFieldFiles as well.
+        fields = collapse(fields, base_type=StreamFieldFile)
+
+        # Filter the list down to just StreamFieldFile instances
+        file_fields = (f for f in fields if isinstance(f, StreamFieldFile))
+
+        for field in file_fields:
+            # return the on-disk file path
+            yield field.name
```

Comment:
I feel we should be returning the file descriptor which is an acceptable public API to be working with for files.

Reply:
Fixed and rebased.